vaadin-spring/src/test/java/com/vaadin/flow/spring/scopes/VaadinRouteScopeTest.java

```diff
@@ -150,4 +125,102 @@ public void destroySession_sessionAttributeIsCleanedAndDestructionCallbackIsCall
         verify(factory, times(2)).getObject();
     }
 
+    @Test
+    public void refresh_uiWithTheSameWindowName_beanInScopeIsDestroyedAfterRefresh() {
+        UI ui = mockUI();
+
+        UI anotherUI = makeAnotherUI(ui);
+
+        ExtendedClientDetails details = Mockito
+                .mock(ExtendedClientDetails.class);
+        Mockito.when(details.getWindowName()).thenReturn("bar");
+        ui.getInternals().setExtendedClientDetails(details);
+        anotherUI.getInternals().setExtendedClientDetails(details);
+
+        ui.getSession().addUI(ui);
+        ui.getSession().addUI(anotherUI);
```

Comment:
Add one more UI with another `windowName`. It would check that current UI is replaced by the same window name, and not with another window name, i.e. covers this `windowName.equals(getWindowName(currentUI))`

Reply:
another test is added for this case